### DISCUSSION TITLE
How to use gflags as a meson subproject?

### DISCUSSION THREAD

**Discussion by hlqqlm** [mesonbuild/meson#10052] [Q&A]
I want to use gflags as a meson subprjoect. gflags is similar as gtest, it's also a google's project. Url is <https://github.com/gflags/gflags>. I think it should be similar as gtest, write a wrap file, get patch file from WrapDB, but I am wrong. There is not exist gflags patch file at WrapDb : https://mesonbuild.com/Wrapdb-projects.html. What can I do with this?

**Comment by eli-schwartz** [comment]
Looks like someone will need to write a meson.build for gflags, and perhaps contribute it to the WrapDB. Is that something you would like to take a stab at? Do you want advice/guidance on doing so?

**Reply by hlqqlm** [reply]
I will try this. I am a new hand, could you give some advice/guidance for starting?

**Comment by xnhp0320** [comment]
The below script works:

```meson 
cmake = import('cmake')
gflags_proj = cmake.subproject('gflags')
message('CMake targets:\n - ' + '\n - '.join(gflags_proj.target_list()))
gflags_dep = gflags_proj.dependency('gflags_static')
```
